Clonacion del proyecto y generacion de ramas

fgarcia1105 opened the pull request:
Se crean las ramas de Dev, Main y stack para trabajar en el proyecto designado

Commit:
Clonacion del proyecto y generacion de ramas

JosueQuinonesRivera commented:
This PR should point to your dev branch

jorge-benavides commented:
Fernando, podrias cambiar el PR para que apunte a tu rama `main-fernando-garcia`?

Commit:
- Se crean las Ramas: main-feriado-garcia, dev-fernando-garcia y task-fernando-garcia
- Se modifica el model de datos Movie y se agrega el modelo Results
- Se modifica el consumo de la API
- Se aplican controles de acceso a clases, métodos y propiedades

JosueQuinonesRivera commented on BAZProjectC1/MovieAPI.swift:
```diff
@@ -6,30 +6,33 @@
+            if error != nil {
+                print (error?.localizedDescription ?? "Error")
+                return
             }
-        }
-
-        return movies
+            
+            do{
+                guard let data = data else {
+                    return
+                }
+                
+                let result = try JSONDecoder().decode(Results.self, from: data)
+                completion(result.results)
+            } catch {
+                print("Se produjo el siguiente error: \(error.localizedDescription)")
```
Is not a good practice to keep prints to yourself. The print is not going to be seen by the users in the app. If you strongly need to have it. Use a `debugPrint` instead.
Moreover, if an exception occurs, this code is not going to call the `completion`. The app will stay in an undefined state when having an error (this could be an infinite loader for example).

JosueQuinonesRivera replied:
Consider moving the response handling logic to a separate method to preserve the SOLID principles.

JosueQuinonesRivera commented on BAZProjectC1/TrendingViewController.swift:
```diff
@@ -6,42 +6,39 @@
-        
-        movies = movieApi.getMovies()
-        tableView.reloadData()
+         self.movieAPI.getMovies { movies in
+             self.movies = movies
+            DispatchQueue.main.async {
+                self.tableView.reloadData()
+            }
+        }
```
Consider moving this logic to a separate method.

JosueQuinonesRivera commented on BAZProjectC1/TrendingViewController.swift:
```diff
@@ -6,42 +6,39 @@
     override func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
         var config = UIListContentConfiguration.cell()
-        config.text = movies[indexPath.row].title
+        config.text = "\(indexPath.row).- " + movies[indexPath.row].title
```
Is there any reason to add this number?

JosueQuinonesRivera commented:
Please review the comments and remember that you should open a PR to your dev branch. This is pointing to the main branch for the repo. This PR can't be merged into main
